### PR TITLE
[CSR-4778] Date component update

### PR DESCRIPTION
**Pull request by dylanmau**
Create new date filter for datagrid
update styling for calendar 
removed unused props from datepicker 
deprecate datepopover, will replace this with the new date filter in next. 

Change for next made in https://github.com/TimeChimp/front-end/pull/182

### CODE REVIEW

File: src/components/datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -248,33 +141,24 @@ export const Datepicker = ({
...
+        value={customValue}
+        locale={localeObj}
+        quickSelect={false}
+        overrides={{
+          ...datepickerBaseOverrides,
+          ...overrides,
+        }}
+        {...rest}
+      />
+    </>
```
Comment (hiddeottens) [inline src/components/datepicker/Datepicker.tsx:162]:
```suggestion
      <DatePicker
        value={customValue}
        locale={localeObj}
        quickSelect={false}
        overrides={{
          ...datepickerBaseOverrides,
          ...overrides,
        }}
        {...rest}
      />
```

File: src/components/datepicker/components/calendar/CalendarComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        },
+      },
+    },
+    MonthYearSelectPopover: {
+      props: {
+        overrides: {
+          Body: {
+            style: () => ({
+              ...borderRadius(radius200),
+              zIndex: 99999999,
```
Comment (hiddeottens) [inline src/components/datepicker/components/calendar/CalendarComponent.tsx:44]:
```suggestion
              zIndex: HIGH_Z_INDEX,
```

**Review by hiddeottens — COMMENTED** [review]
Very nice! just some minor comments

File: src/components/date-filter/DateFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    {
+      id: QuickSelectDateOption.YESTERDAY,
+      label: translations?.yesterday ?? 'Yesterday',
+      beginDate: new TcDate().subtract(1, 'day').toDate(),
+      endDate: new TcDate().subtract(1, 'day').toDate(),
+    },
+    {
+      id: QuickSelectDateOption.THIS_WEEK,
+      label: translations?.thisWeek ?? 'This week',
+      beginDate: new TcDate().startOf('week', 1).getDateWithoutTimeAsUTC(),
```
[the remark this answers is not in the document]
Comment (dylanmau) [reply src/components/date-filter/DateFilter.tsx]: Not sure why, think I triggered some autocomplete after copying it from datepicker. Changed it now.

**Review by hiddeottens — APPROVED** [review]
Nice 🚀

**Review by tarashaidarTimechimp — APPROVED** [review]

**Pull request merged**